darshan-util/pydarshan/darshan/experimental/plots/plot_opcounts.py

```diff
@@ -0,0 +1,157 @@
+    # Gather MPIIO
+    if 'MPI-IO' in mods:
+        #mpiio_record = backend.log_get_mpiio_record(log)
+        #mpiio = dict(zip(backend.counter_names("mpiio"), mpiio_record['counters']))
+
+        mpiio = mods['MPI-IO']
+
+        mpiind_vals = [
+            mpiio['MPIIO_INDEP_READS'],
+            mpiio['MPIIO_INDEP_WRITES'],
+            mpiio['MPIIO_INDEP_OPENS'],
+            0, # stat
+            0, # seek
+            0, # mmap
+            0, # sync
+        ]
+
+        mpicol_vals = [
+            mpiio['MPIIO_COLL_READS'],
+            mpiio['MPIIO_COLL_WRITES'],
+            mpiio['MPIIO_COLL_OPENS'],
+            0, # stat
+            0, # seek
+            0, # mmap
+            mpiio['MPIIO_SYNCS']
+        ]
+
+    # Gather Stdio
+    if 'STDIO' in mods:
+        #stdio_record = backend.log_get_stdio_record(log)
+        #stdio = dict(zip(backend.counter_names("STDIO"), stdio_record['counters']))
+
+        stdio = mods['STDIO']
+
+        stdio_vals = [
+            stdio['STDIO_READS'],
+            stdio['STDIO_WRITES'],
+            stdio['STDIO_OPENS'],
+            0, # stat
+            stdio['STDIO_SEEKS'],
+            0, # mmap
+            stdio['STDIO_FLUSHES']
+        ]
+
+
+
+    def as_csv():
+        text = ""
+        text += ','.join(labels) +  ',Layer' + "\n"
+        text += ','.join(str(x) for x in posix_vals) + ',POSIX' + "\n"
+        text += ','.join(str(x) for x in mpiind_vals) + ',MPIIND' + "\n"
+        text += ','.join(str(x) for x in mpicol_vals) + ',MPICOL' + "\n"
+        text += ','.join(str(x) for x in stdio_vals) + ',STDIO' + "\n"
+
+        return text
+
+
+
+    print(as_csv())
+
+
+    x = np.arange(len(labels))  # the label locations
+    width = 0.15  # the width of the bars
+
+    fig, ax = plt.subplots()
+    rects1 = ax.bar(x - width/2 - width, posix_vals, width, label='POSIX')
+    rects2 = ax.bar(x - width/2, mpiind_vals, width, label='MPI-IO Indep.')
+    rects3 = ax.bar(x + width/2, mpicol_vals, width, label='MPI-IO Coll.')
+    rects4 = ax.bar(x + width/2 + width, stdio_vals, width, label='STDIO')
+
+    # Add some text for labels, title and custom x-axis tick labels, etc.
+    ax.set_ylabel('Count')
+    ax.set_title('I/O Operation Counts')
+    ax.set_xticks(x)
+    ax.set_xticklabels(labels)
+    ax.legend()
+
+
+    def autolabel(rects):
+        """Attach a text label above each bar in *rects*, displaying its height."""
+        for rect in rects:
+            height = rect.get_height()
+            ax.annotate(
+                '{}'.format(height),
+                xy=(rect.get_x() + rect.get_width() / 4 + rect.get_width(), height),
+                xytext=(0, 3),  # 3 points vertical offset
+                textcoords="offset points",
+                ha='center', va='bottom', rotation=45
+                )
+
+
+    autolabel(rects1)
+    autolabel(rects2)
+    autolabel(rects3)
+    autolabel(rects4)
+
+    fig.tight_layout()
+
+    #plt.show()
+    
+
+    return plt
```

Comment:
As far as I can tell, the `plt` variable here  has no actual association with any of the figure/axis objects in the function, so I believe you're currently just returning `matplotlib.pyplot` generically.

The plots we're targeting here are pretty simple I think. I had started doing some work on my fork feature branch `treddy_data_access_by_filesystem` and the results were a bit like the one below, though I was never really happy with how it was working.

![sample-dxt-simple darshan_data_access_by_category](https://user-images.githubusercontent.com/7903078/121082450-58e20d80-c79b-11eb-9d66-0706296c9973.png)

I guess I'll have to dig into the aggregator code you've written if that is what Shane et al. want to use. Once it is clear to me that this function is actually working, we could check the output against what Phil was envisioning.

Reply:
This should be moved to https://github.com/darshan-hpc/darshan/pull/397